src/Microsoft.ML.SearchSpace/Attribute/ChoiceAttribute.cs

```diff
@@ -26,7 +26,7 @@ public ChoiceAttribute(params object[] candidates)
             var candidatesType = candidates.Select(o => o.GetType()).Distinct();
             Contracts.Assert(candidatesType.Count() == 1, "multiple candidates type detected");
 
-            Option = new ChoiceOption(candidates.Select(c => Convert.ToString(c, CultureInfo.InvariantCulture)).ToArray());
```

Comment:
>candidates [](http://example.com/codeflow?start=38&length=10)

why you need to call ToArray if `candidates` itself is array? this apply to other change in this file.

Reply:
### Resolved